### ISSUE TITLE
suggestion for all libraries: implement a log-listener and delete the logger

### ISSUE THREAD

**Issue by BradleyScrim** [sbmlteam/jsbml#217]
The `jsbml-core` library implements a logger with log4j. In v1.5 the log4j version is 2.3

Using this in a library in another project with another log-framework or other versions leads to conflicts. Especially if other libraries also bring their own logger.

Implementing a simple abstract LogListener could solve the problem. Everyone using the library has to implement the concrete Listener and could connect it to the used logger framework.

Benefits
- simplify code
- keep dependency tree small
- keep maintainability

**Comment by lcottret** [comment]
I completely agree ! It's very difficult to tune log4j when we use jsbml with other libraries using also log4j.

**Comment by draeger** [comment]
Would it help to replace the log4j dependency by Java's on-board logging mechanism?

**Comment by niko-rodrigue** [comment]
We are staying on log4j 2.3 at the moment as we have not moved yet to required java 1.8. That 's probably something we should do for next release(s) as it prevent us to upgrade several libraries to their latest versions.

**Comment by BradleyScrim** [comment]
> Would it help to replace the log4j dependency by Java's on-board logging mechanism?

Sorry, "**logging facade**" seems to be the right word for what I mean. A [nice short explanation can be found here](https://stackoverflow.com/questions/9740569/logger-for-java-library#9741118).

**Comment by niko-rodrigue** [comment]
It was already my intention to move to the slf4j api and I just noticed they have a migration tool that would help to modify all the log4j call and import.

**Comment by dyrpsf** [comment]
Hi @draeger and @niko-rodrigue ! I have started the migration to the SLF4J facade to resolve the dependency conflict issues mentioned here. 

I've opened two initial PRs to update the dependencies and migrate the loggers in `AbstractSBase` (JSBML) and `SBMLinterpreter` (SBSCL):
* JSBML PR: https://github.com/sbmlteam/jsbml/pull/300
* SBSCL PR: https://github.com/draeger-lab/SBSCL/pull/110

Both compile perfectly with the new `slf4j-api`. Let me know if you'd like me to extend this migration to other specific classes!
